modules/auxiliary/scanner/http/jenkins_login.rb

```diff
@@ -70,4 +68,35 @@ def run_host(ip)
       end
     end
   end
+
+  private
+
+  # This method uses the provided URI to determine whether login is possible for Jenkins.
+  # Based on the contents of the provided URI, the method looks for the login form and
+  # extracts the endpoint used to authenticate against.
+  #
+  # @param [URI, String] target_uri The targets URI
+  # @return [String, nil] URI for successful login
+  def jenkins_uri_check(target_uri, keep_cookies: false)
```

Comment:
has this been duplicated from the jenkins login scanner class?

Reply:
ah it's different to the login scanner class but the same as the method from this mixin Msf::Exploit::Remote::HTTP::Jenkins? (except for `valid_response?(res)` replaceing `res&.code == 200`)